File: frontend/src/models/analyzer/MetaDto.ts
import type { RawZonedDateTime } from "@/models/analyzer/common"

export interface MetaDto {
  name: string;
  createdAt: RawZonedDateTime;
  updatedAt: RawZonedDateTime;
  indexedAt: RawZonedDateTime;
  commitCount: number;
  strandSymbolCount: number;
  symbolCount: number;
  strandCount: number;
}

File: analyzer/src/main/java/com/mategka/dava/analyzer/serialization/model/MetaDto.java
package com.mategka.dava.analyzer.serialization.model;

import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

import java.time.ZonedDateTime;

@Value
@Builder
public class MetaDto {

  @NonNull
  String name;

  @NonNull
  ZonedDateTime createdAt;

  @NonNull
  ZonedDateTime updatedAt;

  @NonNull
  ZonedDateTime indexedAt;

  int commitCount;
  long strandSymbolCount;
  long symbolCount;
  int strandCount;

}
